zookeeper-server/src/main/java/org/apache/zookeeper/server/persistence/FileSnap.java

```diff
@@ -260,6 +260,8 @@ public synchronized void serialize(
                     Util.getZxidFromName(snapShot.getName(), SNAPSHOT_FILE_PREFIX),
                     snapShot.lastModified() / 1000);
             }
+        } else {
+            throw new IllegalStateException("FileSnap has already been closed");
```

Comment:
will existing code handle this RuntimeException properly ?
what about throwing a IOException ?

Reply:
I dunno if I suggested IllegalStateException originally, but it's what I had in mind when I suggested this should throw an exception. 

If you call close() on FileSnap, and it's already closed, that's a programmer's mistake. There *should not* be any handling for this, because the object is in a state which it should never be. The best thing in this case is for the process to crash (this isn't necessarily the case though, due to the catchall Exception handlers in QuorumPeer).

Reply:
@ivankelly @eolivelli This is the serialize() method of FileSnap. Call chain is:
FileSnap.serialize() <- FileTxnSnapLog.save() <- ZooKeeperServer.takeSnapshot()

`takeSnapshot()` catches IOException and call system exit with error code of `TXNLOG_ERROR_TAKING_SNAPSHOT`. Sounds like this should be the expected behavior.

Reply:
Ok, IOException works in that case.